File: src/main/java/com/libraryjava/domain/book/Book.java
package com.libraryjava.domain.book;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
public class Book {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    private BookType type;

    public Book(String name, BookType type) {
        this.name = name;
        this.type = type;
    }

    public static Book fixture(String name, BookType type) {
        return new Book(name, type);
    }
}

File: library/src/main/kotlin/com/library/domain/book/Book.kt
package com.library.domain.book

import jakarta.persistence.*

@Entity
class Book(
    val name: String,

    @Enumerated(EnumType.STRING)
    val type: BookType,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
) {

    init {
        if (name.isBlank()) {
            throw IllegalArgumentException("이름은 비어있을 수 없습니다.")
        }
    }

    // 테스트에 이용할 객체를 만드는 함수 - Object Mother
    // 이렇게 생겨난 테스트용 객체를 Test Fixture 라고 부른다.
    companion object {
        fun fixture (
            name: String = "책 이름",
            type: BookType = BookType.COMPUTER,
            id: Long? = null,
        ): Book {
            return Book(
                name = name,
                type = type,
                id = id,
            )
        }
    }
}